impl/maven-impl/src/test/java/org/apache/maven/impl/DefaultPluginXmlFactoryTest.java

```diff
@@ -0,0 +1,255 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
test only bug and give follow up as low of test and todo:

<img width="1830" alt="image" src="https://github.com/user-attachments/assets/a554a4bc-24f0-465d-bcd4-db476d019b1b" />